src/Riok.Mapperly/Descriptors/MappingBodyBuilders/MemberMappingBuilder.cs

```diff
@@ -84,6 +85,17 @@ public static bool TryBuild(
             return false;
         }
 
+        if (memberMappingInfo.IsSourceNullable && !targetMember.MemberType.IsNullable())
```

Comment:
Couldn't you use `delegateMapping.TargetType.IsNullable()` instead of `!targetMember.MemberType.IsNullable()` which should solve the `ShouldPassNullValueToNullableUserMappingMethod` test.

Reply:
I hope you can forgive me if I am misunderstanding you, I changed it to `!delegateMapping.TargetType.IsNullable()` but that test is still failing, and I added the diagnostic for the NullableMappingBuilder

Reply:
Oh my bad I mixed it up a bit. I cannot try it myself right now, but I think you would need to check source and target... `memberMappingInfo` contains the source/and target member being mapped. `delegateMapping` is the mapping used to map from the source to the target (convert the types). This means nullability could mismatch from source to the source of the delegateMapping and from the target of the delegateMapping to the target member.
So I think the correct condition would be `(!memberMappingInfo.TargetMember.MemberType.IsNullable() && delegateMapping.TargetType.IsNullable())
            || (!delegateMapping.SourceType.IsNullable() && memberMappingInfo.TargetMember.MemberType.IsNullable())`.

Reply:
Everything seems to work as expected with the exception of this test: 
DisabledNullableClassPropertyToNonNullableProperty now gives a diagnostic, don't know if that's desired
And I would like your opinion on the setting to disable the warning. This was requested in the original ticket, but it can be achieved with pragma disable as well ofcourse, it's not a setting like the other settings which actually change how the mappings work. So:
a. Is this something that should be added?
b. If so, make it one setting for both these diagnostics together or separate settings? I would lean to one setting personally

Reply:
Regarding `DisabledNullableClassPropertyToNonNullableProperty`: Unfortunately, at the beginning of Mapperly, we decided to treat nullable-disabled contexts as if they were nullable-enabled contexts with all possible nullable annotations set. In retrospect, this was a bad decision, and we want to change it in the future, but it will take a lot of work. There are scenarios where nullable-disabled needs to be treated as non-nullable (for example here, in queryable projections, ...)...

Regarding the setting: The severity of these diagnostics can be set quite flexibly via editorconfig files (on a per-file or per-directory basis). However, I think enabling or disabling it for a given member is not possible with editorconfig/pragma/SuppressMessageAttribute, so a config is nice. I'd implement this in a separate PR, as this PR already adds a useful feature. My API suggestion for this would be to add a new property `bool SuppressNullMismatchDiagnostic` to the `MapPropertyAttribute`.